src/main/java/edu/escuelaing/ieti/matchmaking/persistence/impl/InMemoryActivityRepository.java

```diff
@@ -0,0 +1,65 @@
+package edu.escuelaing.ieti.matchmaking.persistence.impl;
+
+import edu.escuelaing.ieti.matchmaking.exception.MatchmakingException;
+import edu.escuelaing.ieti.matchmaking.model.Activity;
+import edu.escuelaing.ieti.matchmaking.persistence.ActivityRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Service
+public class InMemoryActivityRepository implements ActivityRepository {
```

Comment:
Have in mind you may need to modify this implementation when the persistence layer gets created.